### PR TITLE
#1316 Fixed bug loading from classpath within .zip files

### PR DESCRIPTION
**Pull request by ernestojpg**
Quick fix for the issue #1316. The method `unpackFromJarURL()` will now take into consideration .zip files in addition to standard .jar files.

I don't know if we should make it even for generic and take also more file extensions into consideration.

### CODE REVIEW

**Comment by vietj** [comment]
Hi, can you add a test for this please ?

**Comment by ernestojpg** [comment]
Sure! Can I update this pull request, or should I close it and create a new one?

**Comment by vietj** [comment]
you can still push a commit on this PR

File: src/main/java/io/vertx/core/impl/FileResolver.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
 import java.io.InputStream;
 import java.io.UnsupportedEncodingException;
 import java.net.URL;
-import java.net.URLClassLoader;
```
Comment (ernestojpg) [inline src/main/java/io/vertx/core/impl/FileResolver.java]: No used import

File: src/main/java/io/vertx/core/impl/FileResolver.java [hunk 2]
```diff
@@ -57,7 +56,7 @@
 
   private static final String DEFAULT_CACHE_DIR_BASE = ".vertx";
   private static final String FILE_SEP = System.getProperty("file.separator");
-  private static boolean NON_UNIX_FILE_SEP = !FILE_SEP.equals("/");
+  private static final boolean NON_UNIX_FILE_SEP = !FILE_SEP.equals("/");
```
Comment (ernestojpg) [inline src/main/java/io/vertx/core/impl/FileResolver.java]: This is really a constant

**Comment by ernestojpg** [comment]
Added some tests

**Comment by vietj** [comment]
thanks!

**Pull request merged**